apps/google-analytics-4/frontend/src/hooks/useSidebarSlug/useSidebarSlug.tsx

```diff
@@ -5,22 +5,35 @@ import { ContentEntitySys, SidebarExtensionSDK } from '@contentful/app-sdk';
 import { pathJoin } from 'utils/pathJoin';
 import useGetFieldValue from '../useGetFieldValue';
 
+const SLUG_FIELD_INPUT_DELAY = 500;
+
 export const useSidebarSlug = (slugFieldInfo: ContentTypeValue) => {
-  const [isPublished, setIsPublished] = useState(false);
   const sdk = useSDK<SidebarExtensionSDK>();
 
   const { slugField, urlPrefix } = slugFieldInfo;
   const slugFieldValue = useGetFieldValue(slugField);
 
+  const [isPublished, setIsPublished] = useState(false);
+  const [debouncedSlugFieldValue, setDebouncedSlugFieldValue] = useState(slugFieldValue);
+
   const handlePublishedStatus = (sys: ContentEntitySys) => {
     setIsPublished(Boolean(sys.publishedAt));
   };
 
+  useEffect(() => {
+    const timeout = setTimeout(
```

Comment:
ProTip: you can use [lodash debounce](https://lodash.com/docs/#debounce) if you want to avoid roll-your-own :)